### ISSUE TITLE
componentWillReceiveProps raises warning

### ISSUE THREAD

**Issue by frankyjuang** [yahoo/react-stickynode#128]
Reference: https://fb.me/react-unsafe-component-lifecycles

```
Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the 
```

**Comment by ahahn95** [comment]
@frankyjuang Seeing the same behavior when updating my React version. Did you find a solution?

**Comment by frankyjuang** [comment]
@ahahn95 nope. I guess there's no way around but fix the library itself.

**Comment by gugiserman** [comment]
I just upgraded React in a couple of projects of mine and this is the last library still popping up warnings. It would be fairly easy to add the `UNSAFE` prefix but it's not a good long-term fix, right?

Is there any way to ignore this error temporarily so we don't have our consoles full of yellow blocks for now at least?

**Comment by maxreality** [comment]
UNSAFE will be supported throughout 17. Facebook states they have over 50k components and don't plan to end support for the lifecycle hooks instantly. However, given the activity in this repo, I think it would be wise to move your projects away from this library. In the interim, you could either try submitting a PR yourself or modifying the library and ensuring it isn't rewritten on the next npm install (dirty).

**Comment by redonkulus** [comment]
This was fixed in https://github.com/yahoo/react-stickynode/pull/124 and released in v3.0.0
